src/main/java/com/dev/cinema/dao/AbstractDao.java

```diff
@@ -0,0 +1,100 @@
+package com.dev.cinema.dao;
+
+import com.dev.cinema.exceptions.DataProcessingException;
+import java.util.List;
+import java.util.Optional;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public abstract class AbstractDao<T> {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
+    protected AbstractDao(SessionFactory sessionFactory) {
```

Comment:
https://www.baeldung.com/spring-autowired-abstract-class#constructor-injection
please see this article